src/access/DicomAccess.js

```diff
@@ -0,0 +1,22 @@
+// Abstract base class for DICOM access implementations
+export class DicomAccess {
+  static async createInstance(url, options) {
+    if (url.startsWith("http")) {
```

Comment:
You need to have at least two implementations - local dicomweb file system and dicomweb http as this should work for both types.